library/src/main/java/com/novoda/downloadmanager/DownloadFile.java

```diff
@@ -47,9 +47,9 @@
     }
 
     void download(final Callback callback) {
-        callback.onUpdate(downloadFileStatus);
+        downloadFileStatus.markAsDownloading();
```

Comment:
@zegnus I have removed the check that was marking files as `DOWNLOADING` when it was either `QUEUED` or `ERROR`. As far as I am aware we should always be able to mark them as `DOWNLOADING` initially when calling `download`, the status will be immediately updated after this point with the file size checks etc.

Reply:
It's important to note this will currently retrigger a download if the file size is incorrect, which is fixed by #317